client/client-lib/src/lib.rs

```diff
@@ -90,6 +91,18 @@ pub struct PaymentParameters {
     pub maybe_internal: bool,
 }
 
+// Placeholder struct for identifying federations across clients
+#[derive(Debug, Serialize, Deserialize, Clone, Eq, Hash, PartialEq)]
+pub struct FederationId(pub String);
+
+impl FromStr for FederationId {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        Ok(FederationId(s.to_string()))
+    }
+}
+
```

Comment:
We already have this in the gateway code, right? Can we remove it there if it's now in the client lib?